lib/pure/collections/sequtils.nim

```diff
@@ -497,9 +497,10 @@ template toSeq*(iter: untyped): untyped =
   ##   assert odd_numbers == @[1, 3, 5, 7, 9]
 
   when compiles(iter.len):
+    let s = iter
```

Comment:
My only concern here is that if `iter` is an object (say, a table) the assignment will copy the entire structure, which may be quite expensive.